kube-client/src/client/middleware/refresh_token.rs

```diff
@@ -1,197 +0,0 @@
-use std::{
-    pin::Pin,
-    task::{Context, Poll},
-};
-
-use futures::{ready, Future};
-use http::{header::AUTHORIZATION, Request, Response};
-use pin_project::pin_project;
-use tower::{layer::Layer, BoxError, Service};
-
-use crate::client::auth::RefreshableToken;
-
-/// `Layer` to decorate the request with `Authorization` header with refreshable token.
-/// Token is refreshed automatically when necessary.
-pub struct RefreshTokenLayer {
-    refreshable: RefreshableToken,
-}
-
-impl RefreshTokenLayer {
-    pub(crate) fn new(refreshable: RefreshableToken) -> Self {
-        Self { refreshable }
-    }
-}
-
-impl<S> Layer<S> for RefreshTokenLayer {
-    type Service = RefreshToken<S>;
-
-    fn layer(&self, service: S) -> Self::Service {
-        RefreshToken {
-            refreshable: self.refreshable.clone(),
-            service,
-        }
-    }
-}
-
-pub struct RefreshToken<S> {
```

Comment:
Apparently these have been public, indirectly. Changing this to `pub(crate)` fails.
I guess I don't understand how `pub` in private modules work.

Reply:
That's weird. It doesn't show up on docs.rs: https://docs.rs/kube-client/latest/kube_client/?search=Refresh

Reply:
But I think this might be the case still. There are some weird cases where a pub thing can leak something that's not pub.

Reply:
The new `pub`s are not shown in docs either even without `#[doc(hidden)]`. I don't think it's possible to access them from outside, and `#![deny(missing_docs)]` doesn't flag them, either.
`#[doc(hidden)]` might be misleading, so I'll remove them and add a comment explaining why it's `pub`.

Reply:
`pub` in private modules work more or less like in public modules, except they aren't reachable by their "natural" path (`crate::declaring_module::Type`). They can still be used publically if they have a reachable alternate path (via `pub use` or associated types).